python/looker_sdk/error.py

```diff
@@ -21,33 +21,128 @@
-        documentation_url: documentation link
+    message: main error info message
+    errors: array of error details
+    documentation_url: documentation link
+    error_doc_url: URL that may point to additional useful information
+    error_doc: Markdown doc that may contain additional useful information
     """
 
     message: str
-    errors: Optional[Sequence[ErrorDetail]] = attr.ib(default=[], kw_only=True)
-    documentation_url: Optional[str] = attr.ib(default="", kw_only=True)
+    errors: Sequence[ErrorDetail] = attr.ib(default=[], kw_only=True)
+    documentation_url: str = attr.ib(default="", kw_only=True)
+    error_doc_url: str = ""
+    error_doc: str = ""
+
+    def __str__(self):
+        sep = "****\n"
+        return f"""
+    message:           {self.message}
+    documentation_url: {self.documentation_url}
+    error_doc_url:     {self.error_doc_url}
+    error details:
+    {sep.join(self.errors)}
+    """
+
+
+"""Error Doc Helper class
+"""
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class ErrorDocHelper:
+    """Error Doc Helper:
+    error_doc_url: link
+    """
+
+    ERROR_CODES_URL: str = "https://static-a.cdn.looker.app/errorcodes/"
+    lookup_dict: Dict[str, Dict[str, str]] = {}
+    RE_PATTERN: str = (
+        """(https://docs\.looker\.com/r/err/|https://cloud\.google\.com/looker/docs/r/err/)(.*)/(\d{3})(.*)"""
+    )
+    pattern = re.compile(RE_PATTERN)
+
+    def get_index(self, url: str = ERROR_CODES_URL) -> None:
+        r = requests.get(f"{url}index.json")
+        self.lookup_dict = json.loads(r.text)
+
+    def lookup(
+        self, url: str = ERROR_CODES_URL, code: str = "", path: str = ""
+    ) -> Tuple[str, str]:
+        if len(self.lookup_dict) == 0:
+            self.get_index(url=url)
+
+        error_doc_url: str = ""
+        error_doc: str = ""
+        if path:
+            try:
+                error_doc_url = self.lookup_dict[f"{code}{path}"]["url"]
+            except KeyError:
+                error_doc = f"### No documentation found for {code}{path}"
+        if not error_doc_url:
+            try:
+                error_doc_url = self.lookup_dict[code]["url"]
+            except KeyError:
+                if not error_doc:
+                    error_doc = f"### No documentation found for {code}"
+
+        if error_doc_url:
+            r = requests.get(f"{self.ERROR_CODES_URL}{error_doc_url}")
+            error_doc = r.text
+
+        return (f"{self.ERROR_CODES_URL}{error_doc_url}", error_doc)
+
+    def parse_and_lookup(
+        self, error_url: str, url: str = ERROR_CODES_URL
+    ) -> Tuple[str, str]:
+        m = re.search(self.RE_PATTERN, error_url)
+        if not m:
+            return ("", "")
+
+        code: str = cast(Tuple[str, str, str, str], m.groups())[2]
+        path: str = cast(Tuple[str, str, str, str], m.groups())[3]
+        try:
+            return self.lookup(url=url, code=code, path=path)
+        except requests.exceptions.RequestException:
```

Comment:
@jkaster RequestException is the parent exception of everything in the requests library. If anything goes wrong it is caught here.